src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -155,37 +156,47 @@ public bool ValidateProcessedBlock(Block processedBlock, TxReceipt[] receipts, B
 
     private bool ValidateWithdrawals(Block block, IReleaseSpec spec, out string? error)
     {
-        if (spec.WithdrawalsEnabled && block.Withdrawals is null)
+        static bool Fail(ILogger logger, string error)
         {
-            error = $"Withdrawals cannot be null in block {block.Hash} when EIP-4895 activated.";
-
-            if (_logger.IsWarn) _logger.Warn(error);
-
+            if (logger.IsWarn) logger.Warn(error);
             return false;
         }
 
-        if (!spec.WithdrawalsEnabled && block.Withdrawals is not null)
+        if (spec.WithdrawalsEnabled && block.Withdrawals is null)
         {
-            error = $"Withdrawals must be null in block {block.Hash} when EIP-4895 not activated.";
-
-            if (_logger.IsWarn) _logger.Warn(error);
+            return Fail(_logger, error = $"Withdrawals cannot be null in block {block.ToString(Block.Format.FullHashAndNumber)} when EIP-4895 activated.");
+        }
 
-            return false;
+        if (!spec.WithdrawalsEnabled && block.Withdrawals is not null)
+        {
+            return Fail(_logger, error = $"Withdrawals must be null in block {block.ToString(Block.Format.FullHashAndNumber)} when EIP-4895 not activated.");
         }
 
         if (block.Withdrawals is not null)
         {
             if (!ValidateWithdrawalsHashMatches(block, out Keccak withdrawalsRoot))
             {
-                error = $"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}";
-                if (_logger.IsWarn) _logger.Warn($"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}");
+                return Fail(_logger, error = $"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}");
+            }
 
-                return false;
+            // We should initialize it from parent?
+            ulong expectedIndex = block.Withdrawals.FirstOrDefault()?.Index ?? 0;
```

Comment:
If we decide to validate indexes, we should validate amount != 0 for sure

Reply:
We do [GetPooledTransactionsMessage](https://github.com/NethermindEth/nethermind/pull/5123/files#diff-e3b880e83ad43a7ae2bb98490f625d74598e445fab37ea418ea354fcb633de51R187)